plugins/coriolis.py

```diff
@@ -13,13 +22,101 @@
 
 config.delete('shipyard', suppress=True)
 
+logger = get_main_logger()
+
+DEFAULT_NORMAL_URL = 'https://coriolis.io/import?data='
+DEFAULT_BETA_URL = 'https://beta.coriolis.io/import?data='
+DEFAULT_OVERRIDE_MODE = 'auto'
+
+normal_url = ''
+beta_url = ''
+override_mode = ''
+
+normal_textvar = tk.StringVar()
+beta_textvar = tk.StringVar()
+override_textvar = tk.StringVar()
+
+
+def plugin_start3(_) -> str:
+    """Set up URLs."""
+    global normal_url, beta_url, override_mode
+    normal_url = config.get_str('coriolis_normal_url', default=DEFAULT_NORMAL_URL)
+    beta_url = config.get_str('coriolis_beta_url', default=DEFAULT_BETA_URL)
+    override_mode = config.get_str('coriolis_overide_url_selection', default=DEFAULT_OVERRIDE_MODE)
+
+    normal_textvar.set(value=normal_url)
+    beta_textvar.set(value=beta_url)
+    override_textvar.set(value=override_mode)
 
-def plugin_start3(_):
     return 'Coriolis'
 
 
-def shipyard_url(loadout, is_beta):
-    """Return a URL for the current ship"""
+def plugin_prefs(parent: tk.Widget, cmdr: str, is_beta: bool) -> tk.Frame:
+    """Set up plugin preferences."""
+    PADX = 10  # noqa: N806
+
+    conf_frame = nb.Frame(parent)
+    conf_frame.columnconfigure(index=1, weight=1)
+    cur_row = 0
+    nb.Label(conf_frame, text=_(
+        'Set the URL to use with coriolis.io ship loadouts. Note that this MUST end with "/import?data="\n'
+    )).grid(sticky=tk.EW, row=cur_row, column=0, columnspan=2)
+    cur_row += 1
+
+    nb.Label(conf_frame, text=_('Normal URL')).grid(sticky=tk.W, row=cur_row, column=0, padx=PADX)
+    nb.Entry(conf_frame, textvariable=normal_textvar).grid(sticky=tk.EW, row=cur_row, column=1, padx=PADX)
+    cur_row += 1
+
+    nb.Label(conf_frame, text=_('Beta URL')).grid(sticky=tk.W, row=cur_row, column=0, padx=PADX)
+    nb.Entry(conf_frame, textvariable=beta_textvar).grid(sticky=tk.EW, row=cur_row, column=1, padx=PADX)
+    cur_row += 1
+
```

Comment:
These each need a 'Default' button on the right hand side to make it *easy* to revert.

Reply:
Added